### ISSUE TITLE
Template Error in Home Assistant

### ISSUE THREAD

**Issue by dd310** [fmartinou/whats-up-docker#307]
Hi

I noticed that, after configuring WUD with MQTT and Home Assistant, the following warnign appears in the logs:

```
Logger: homeassistant.helpers.template
Source: helpers/template.py:735
First occurred: 16:00:17 (2 occurrences)
Last logged: 16:00:26

Template variable warning: 'dict object' has no attribute 'result_tag' when rendering '{% if value_json.update_kind_kind == "digest" %}{{ value_json.result_digest[:15] }}{% else %}{{ value_json.result_tag }}{% endif %}'
```

Any ideas on how to resolve?
Thanks

**Comment by a-neagoe** [comment]
if you look at the code it looks like the else from the conditionals should extract some tag, store it in a dict and render it with the template.py (flask proably) and the tag is probably not there . I would check very carefully the tags/labels configured for your containers.

**Comment by dd310** [comment]
I tried adding the labels `wud.tag.include: "latest"` and `wud.watch.digest: "true"` to every container, but the issue is still there.
I wouldn't know what else I could try.

**Comment by fenty17** [comment]
@dd310 - are you using custom icons? (Si, fas etc.). I was trying to do this but they've never appeared correctly in Home Assistant so just left it. All my labels are like this:
````
    labels:
      - wud.display.name=iCloudPD
      - wud.display.icon=si:icloud

````

**Comment by dd310** [comment]
> @dd310 - are you using custom icons? (Si, fas etc.). I was trying to do this but they've never appeared correctly in Home Assistant so just left it. All my labels are like this:
> ````
>     labels:
>       - wud.display.name=iCloudPD
>       - wud.display.icon=si:icloud
> 
> ````

No, I only use the display.name tag, never tried the other one.

**Comment by fmartinou** [comment]
Hi @dd310 ,

Can you inspect the `wud` mqtt topics to see how looks like the MQTT messages ?
(`wud/container/...`)

**Comment by dd310** [comment]
![Screenshot_20230804_182616_Home Assistant.jpg](https://github.com/fmartinou/whats-up-docker/assets/17815448/4b866ec1-6370-476f-a229-e051dfbf5f3a)

I hope the screenshot is readable.

**Comment by dd310** [comment]
![Screenshot_20230804_182616_Home Assistant.jpg](https://github.com/fmartinou/whats-up-docker/assets/17815448/4b866ec1-6370-476f-a229-e051dfbf5f3a)

I hope the screenshot is readable. 

EDIT:

Checking the "received" time, this message has probably something to do with the error:

![Screenshot_20230804_183742_Home Assistant.jpg](https://github.com/fmartinou/whats-up-docker/assets/17815448/279503cc-91b9-4d7a-9028-4bcc826e64f2)

**Comment by aheath70** [comment]
Also got this issue. Also running WUD. I don't run any labels as I have just started using WUD and haven't quite got my head around it yet. I just have it watching all containers on all my docker instances.

**Comment by SNoof85** [comment]
FYI I had this issue too.

Working solution here : 
- delete MQTT devices of WUD in HA
- Delete all MQTT messages from WUD from Mosqitto
- Rescan for updates in WUD
- Enjoy !

I has some old containers that were still in HA as unaivailable and in MQTT... Maybe that cleanup did the trick :)

**Comment by rodrigosiviero** [comment]
> FYI I had this issue too.
> 
> Working solution here :
> 
> * delete MQTT devices of WUD in HA
> * Delete all MQTT messages from WUD from Mosqitto
> * Rescan for updates in WUD
> * Enjoy !
> 
> I has some old containers that were still in HA as unaivailable and in MQTT... Maybe that cleanup did the trick :)

Tried this, no luck. Still being flooded by the errors.

**Comment by msaphoto06** [comment]
same problem

**Comment by mudape** [comment]
When there is an error_message there is no result_tag attribute, making the template fail

![image](https://github.com/user-attachments/assets/3c19d88b-e53a-435f-b2e2-fef60db0e980)

https://github.com/getwud/wud/blob/bde9dc6d034cdff28a579f9d4ecdfe0b12992865/app/triggers/providers/mqtt/Hass.js#L15

Change it so a value is returned?
Like `{% else %}{{ value_json.result_tag if value_json.result_tag is defined else value_json.image_tag_value }}{% endif %}`

**Comment by FRvanderVeen** [comment]
I've the same issue with the same reason, namely too much requests.

I dived into the code and I think the result_tag is added here (method `watchContainer` in Docker.js:

https://github.com/getwud/wud/blob/7e579d2b4f64ddc2a4fea184dba13f298dcb238e/app/watchers/providers/docker/Docker.js#L498-L509

When finding the new version succeeds we will have the result.tag if it fails we won't. Doesn't it make sense to add it also when it can't be retrieved, but then with value `unknown`. And probably do the same for the link.

**Comment by headtr1ck** [comment]
I also get the same warning.

**Comment by romlisrl** [comment]
I had same issue.

```
Logger: homeassistant.helpers.template
Source: helpers/template/__init__.py:1757
First occurred: 21:00:45 (1 occurrence)
Last logged: 21:00:45

Template variable warning: 'dict object' has no attribute 'result_tag' when rendering '{% if value_json.update_kind_kind == "digest" %}{{ value_json.result_digest[:15] }}{% else %}{{ value_json.result_tag }}{% endif %}'
```
